FastBank.Services/BankAccountService/BankAccountService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using FastBank.Domain;
+using FastBank.Domain.RepositoryInterfaces;
+using FastBank.Infrastructure.Repository;
+
+namespace FastBank.Services.BankAccountService
+{
+    public class BankAccountService : IBankAccountService
+    {
+        private readonly IBankAccountRepository bankAccountRepository;
+
+        public BankAccountService()
+        {
+            bankAccountRepository = new BankAccountRepository();
+        }
+        public BankAccount? GetBankAccount(Customer customer)
+        {
+            return bankAccountRepository.GetBankAccountByCustomer(customer);
+        }
+
+        public void Add(Customer customer, decimal amount)
+        {
+            if (GetBankAccount(customer) == null)
```

Comment:
GetBankAccount(customer) save the result from that method in a variable and use it in the if statement